de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/metadata/MetadataUtil.java

```diff
@@ -0,0 +1,99 @@
+package org.iplantc.de.diskResource.client.presenters.metadata;
+
+import org.iplantc.de.client.models.avu.Avu;
+import org.iplantc.de.client.models.avu.AvuAutoBeanFactory;
+import org.iplantc.de.client.models.diskResources.MetadataTemplateAttribute;
+import org.iplantc.de.client.models.diskResources.MetadataTemplateAttributeType;
+import org.iplantc.de.client.models.ontologies.OntologyAutoBeanFactory;
+import org.iplantc.de.client.models.ontologies.OntologyLookupServiceQueryParams;
+import org.iplantc.de.diskResource.client.presenters.metadata.proxy.AstroThesaurusLoadConfig;
+import org.iplantc.de.diskResource.client.presenters.metadata.proxy.AstroThesaurusProxy;
+import org.iplantc.de.diskResource.client.presenters.metadata.proxy.MetadataTermLoadConfig;
+import org.iplantc.de.diskResource.client.presenters.metadata.proxy.MetadataTermSearchProxy;
+import org.iplantc.de.diskResource.client.presenters.metadata.proxy.OntologyLookupServiceLoadConfig;
+import org.iplantc.de.diskResource.client.presenters.metadata.proxy.OntologyLookupServiceProxy;
+import org.iplantc.de.diskResource.client.views.search.MetadataTermSearchField;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
```

Comment:
https://stackoverflow.com/questions/6382473/singleton-in-gwt-project. You might want to check this thread about using Singleton in GWT.

Reply:
Thanks!  I looked at the link and I think I'm implementing this singleton correctly so we can also use dependency injection, but if I've missed something please let me know.  The issue seemed to be regarding the use of `new SingletonClass()`, but I'm not doing that anywhere.